api/processors/group.py

```diff
@@ -1,184 +1,14 @@
-        coordinators = Coordinator.objects
-
-        coordinators.filter_below_threshold().update(
-            status=CoordinatorStatus.WORKING_BELOW_THRESHOLD, status_since=timestamp
-        )
-        coordinators.filter_above_threshold_and_within_limit().update(
-            status=CoordinatorStatus.WORKING_OK, status_since=timestamp
-        )
-        coordinators.filter_limit_reached().update(
-            status=CoordinatorStatus.WORKING_LIMIT_REACHED, status_since=timestamp
-        )
-
-    @staticmethod
-    def _set_students_status_start(group: Group, timestamp: datetime.datetime) -> None:
-        group.students.update(
-            status=StudentStatus.STUDYING,
-            status_since=timestamp,
-        )
-
-    @staticmethod
-    def _set_students_status_abort(group: Group, timestamp: datetime.datetime) -> None:
-        group.students.update(
-            status=StudentStatus.AWAITING_OFFER,
-            status_since=timestamp,
-        )
+class GroupProcessor:
+    """A Facade class providing access to actions with groups."""
 
     @staticmethod
-    def _move_related_people_to_former(group: Group) -> None:
-        teachers_current, students_current, coordinators_current = (
-            group.teachers,
-            group.students,
-            group.coordinators,
-        )
-
-        group.teachers.clear()
-        group.students.clear()
-        group.coordinators.clear()
-        group.teachers_former.add(*teachers_current.all())
-        group.students_former.add(*students_current.all())
-        group.coordinators_former.add(*coordinators_current.all())
-        group.save()
-
-    @staticmethod
-    def _set_teachers_status_start(timestamp: datetime.datetime) -> None:
-        teachers = Teacher.objects
-
-        teachers.filter_can_take_more_groups().update(
-            status=TeacherStatus.TEACHING_ACCEPTING_MORE,
-            status_since=timestamp,
-        )
-
-        teachers.filter_cannot_take_more_groups().update(
-            status=TeacherStatus.TEACHING_NOT_ACCEPTING_MORE,
-            status_since=timestamp,
-        )
+    def start(group: Group) -> None:
+        GroupStartProcessor(group=group).process()
 
     @staticmethod
-    def _set_teachers_status_abort(timestamp: datetime.datetime) -> None:
-        teachers = Teacher.objects
-
-        teachers.filter_has_no_groups().update(
-            status=TeacherStatus.AWAITING_OFFER,
-            status_since=timestamp,
-        )
-
-        teachers.filter_has_groups().filter_can_take_more_groups().update(
-            status=TeacherStatus.TEACHING_ACCEPTING_MORE,
-            status_since=timestamp,
-        )
-
-        teachers.filter_cannot_take_more_groups().update(
-            status=TeacherStatus.TEACHING_NOT_ACCEPTING_MORE,
-            status_since=timestamp,
-        )
+    def abort(group: Group) -> None:
+        GroupAbortProcessor(group=group).process()
```

Comment:
i think its a nice approach, nice and concise and tells you where to look for implementation details 👍